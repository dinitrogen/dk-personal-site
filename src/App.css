.bar {
  position: relative;
  overflow: hidden;
}

.bar::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  clip-path: polygon(0 0, 100% 0%, 100% 35%, 0 50%);
}

@media screen and (min-width: 768px) {
  .bar::before {
    clip-path: polygon(0 0, 100% 0%, 100% 35%, 0 60%);
  }
}
