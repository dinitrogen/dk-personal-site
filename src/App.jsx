import './App.css'
import Navbar from './components/Navbar';
import About from './components/About';
import Skills from './components/Skills';
import Projects from './components/Projects';
import { sharedText } from './assets/shared-text';

import { useRef } from 'react';

function App() {

  const aboutRef= useRef();
  const projectsRef = useRef();
  const skillsRef = useRef();
  const contactRef = useRef();

  return (
    <div>
      
        <Navbar aboutRef={aboutRef} projectsRef={projectsRef} skillsRef={skillsRef} contactRef={contactRef} />
        <div className='h-96 bg-cyan-600'></div>
        <div className='h-96 bg-cyan-600'></div>
        <About aboutRef={aboutRef} title="About me" bodyText={sharedText.aboutMe.body} />
        <Projects projectsRef={projectsRef} title="Projects" projects={sharedText.projects} />

        <Skills skillsRef={skillsRef} title="Skills"/>
        <About aboutRef={contactRef} title="Contact"/>
        
        

      
    </div>
  )
}

export default App
