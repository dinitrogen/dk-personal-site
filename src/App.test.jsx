import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "./App";
import userEvent from "@testing-library/user-event";

describe("something truthy and falsy", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("false to be false", () => {
    expect(false).toBe(false);
  });
});

describe("App component", () => {
  it("renders correct heading", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
    expect(screen.getByRole("heading").textContent).toMatch(/hello world/i);
  });

  it("renders goodbye world after click", async () => {
    const user = userEvent.setup();

    const { container } = render(<App />);

    const button = screen.getByRole("button", { name: "Click" });
    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/goodbye world/i);
    expect(container).toMatchSnapshot();
  });
});
