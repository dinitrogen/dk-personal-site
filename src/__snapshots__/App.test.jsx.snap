// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correct heading 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <button
    type="button"
  >
    Click
  </button>
</div>
`;

exports[`App component > renders goodbye world after click 1`] = `
<div>
  <h1>
    Goodbye World
  </h1>
  <button
    type="button"
  >
    Click
  </button>
</div>
`;
